pyfstat/utils/sft.py

```diff
@@ -175,3 +178,124 @@ def get_official_sft_filename(
+    label: Optional[str] = None,
+    quantity: Optional[str] = "power",
+    sqrtSX: Optional[float] = None,
+    fMin: Optional[float] = None,
+    fMax: Optional[float] = None,
+    constraints: Optional[lalpulsar.SFTConstraints] = None,
+    **kwargs,
+) -> matplotlib.axes.Axes:
+    """
+    Compute spectrograms of a set of SFTs.
+    This is useful to produce visualizations of the Doppler modulation of a CW signal.
+
+    Parameters
+    ----------
+    sftfilepattern:
+        Pattern to match SFTs using wildcards (`*?`) and ranges [0-9];
+        multiple patterns can be given separated by colons.
+    detector:
+        Name of the detector of the data that will be plot.
+    savefig:
+        If True, save the figure in `outdir`.
+        If False, return an axis object without saving to disk.
+    outdir:
+        Output folder.
+    label:
+        Output filename.
+    quantity:
+        Magnitude to be plotted.
+        It can be "power" for power, "normpower" for normalized power, "Re" for the real part of
+        the SFTs, and "Im" for the imaginary part of the SFTs.
+        Set to "power" by default.
+    sqrtSX:
+        Amplitude spectral density of the data.
+        Only needed if `quantity = "normpower".
+    fMin, fMax:
+        Restrict frequency range to `[fMin, fMax]`.
+        If None, retrieve the full frequency range.
+    constraints:
+        Constraints to be fed into XLALSFTdataFind to specify detector,
+        GPS time range or timestamps to be retrieved.
+    kwarg: dict
+        Other kwargs, only used to be passed to `matplotlib.pcolormesh`.
+
+    Returns
+    -------
+    ax: matplotlib.axes.Axes
+        The axes object containing the plot.
+    """
+
+    logger.info("Loading SFT data")
+    frequency, timestamps, fourier_data = get_sft_as_arrays(
+        sftfilepattern, fMin, fMax, constraints
+    )
+
+    if savefig:
+        if label is None:
+            raise ValueError("Label needed to save the figure")
+        else:
+            plotfile = os.path.join(outdir, label + ".png")
+            logger.info(f"Plotting to file: {plotfile}")
+
+    plt.rcParams["axes.grid"] = False  # turn off the gridlines
+    fig, ax = plt.subplots(figsize=(0.8 * 16, 0.8 * 9))  # FIXME: make configurable
+    ax.set(xlabel="Time [days]", ylabel="Frequency [Hz]")
+
+    if detector not in timestamps:
+        raise ValueError(
+            f"Detector {detector} not found in timestamps, available detectors are {timestamps.keys()}"
+        )
+
+    time_in_days = (timestamps[detector] - timestamps[detector][0]) / 86400
+
+    if "power" in quantity:
+        logger.info("Computing SFT power")
+        q = fourier_data[detector].real ** 2 + fourier_data[detector].imag ** 2
+        if quantity == "normpower":
+            if sqrtSX is None:
+                raise ValueError(
+                    "Value of sqrtSX needed to compute the normalized power."
+                )
+            Tsft = 1800  # CHANGE!!
```

Comment:
This is how we read it out of the files in https://github.com/PyFstat/PyFstat/blob/master/pyfstat/make_sfts.py#L274 :
```
multi_sft_catalog = lalpulsar.GetMultiSFTCatalogView(
            lalpulsar.SFTdataFind(self.noiseSFTs, SFTConstraint)
        )
for ifo_catalog in multi_sft_catalog.data:
    Tsft = int(round(1.0 / ifo_catalog.data[0].header.deltaF))
```
(here, the loop can be collapsed into a single call)